app/gui/view/debug_scene/cloud-dashboard/src/lib.rs

```diff
@@ -0,0 +1,680 @@
+                const LEN: usize = mem::variant_count::<Self>();
+
+                /// Returns the [`Columns`] variant corresponding to the given discriminant value.
+                ///
+                /// If the discriminant is out of bounds, returns `None`.
+                ///
+                /// Example:
+                /// ```rust
+                /// # use debug_scene_cloud_dashboard::projects_table::Columns;
+                /// assert_eq!(Columns::from_discriminant(0), Some(Columns::Projects));
+                /// assert_eq!(Columns::from_discriminant(1), Some(Columns::LastModified));
+                /// ```
+                pub fn from_discriminant(discriminant: usize) -> Option<Self> {
+                    // It is non-trivial to write a macro that can use `match` for this purpose,
+                    // because a macro can't return both the pattern and the expression parts of a
+                    // `match` arm (i.e., `x => y`). So instead we use an iterator over the values
+                    // `0..` combined with `if` statements. This is functionally equivalent and is
+                    // optimized by the compiler to the same instructions.
+                    let mut i = 0..;
+                    $(if discriminant == i.next().unwrap() { Some(Self::$column_name) } else)*
+                    { unreachable!() }
+                }
+            }
+        };
+    }
+
+    define_columns_enum!(Projects, LastModified, SharedWith, Labels, DataAccess, UsagePlan);
+
+
+    // === Trait `impl`s ===
+
+    impl Display for Columns {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            let debug_str = format!("{:?}", self);
+            let display_str = view::separate_pascal_case_str_with_spaces(&debug_str);
+            write!(f, "{display_str}")
+        }
+    }
+
+
+
+    // =====================
+    // === ProjectsTable ===
+    // =====================
+
+    /// A table of [`Project`]s displayed in the Cloud dashboard.
+    ///
+    /// The table contains information about the [`Project`]s (e.g. their names, their state, etc.)
+    /// as well as components for interacting with the [`Project`]s (e.g. buttons to start/stop) the
+    /// projects, etc.
+    ///
+    /// Under the hood, we use a scrollable grid since the user may have more [`Project`]s than can
+    /// fit on the screen. We use a grid with headers since the user needs to know which [`Project`]
+    /// property each column represents.
+    ///
+    /// [`Project`]: ::enso_cloud_view::project::Project
+    pub type ProjectsTable =
+        ensogl_grid_view::simple::SimpleScrollableSelectableGridViewWithHeaders;
+
+
+
+    // ================
+    // === Position ===
+    // ================
+
+    /// The row and column coordinates of a cell in the [`ProjectsTable`].
+    #[derive(Clone, Copy, Debug, Default)]
+    pub struct Position {
+        row:    Row,
+        column: Col,
+    }
```

Comment:
@farmaazon I suggested that because in the code there was `.0` and `.1` in many places. It's a little bit hard to understand and its way easier to read code if you have `.row` and `.column`. Do you agree with this?